### ISSUE TITLE
Ghost notes: Ghost icon inside piano roll.

### ISSUE THREAD

**Issue by Gabrielxd195** [LMMS/lmms#5267]
Currently to set a piano roll pattern as a ghost, you have to do it from the song editor by clicking left on that pattern and executing the option `` set as ghost in piano-roll ''. good!!.
![Establecer el patron fantasma](https://user-images.githubusercontent.com/26127693/67255133-89eabe80-f44e-11e9-94f0-1da77537e161.png)
The problem is that when you modify the same pattern several times (either to make a correction or modify a note), you have to look for it and set it as many times as necessary through that menu option. This sometimes becomes a bit repetitive.

It would be useful to have this option but within the piano roll, (since basically this option `` photographs '' the written notes). something like:
![fantasma dentro de piano roll](https://user-images.githubusercontent.com/26127693/67255640-15654f00-f451-11e9-8e08-a41d9886e4cb.png)
In this way we simplify that task, and every time you make a modification to the pattern you would only have to press that button without leaving the piano roll. Regards.

**Comment by Spekular** [comment]
Are you asking for a way to show/hide the ghost notes in piano roll? If so
that should be one toggleable button rather than two.

**Comment by Spekular** [comment]
Ah wait, you mean you want to set the currently opened pattern as ghost notes, don't you? I still think that could be a single button, though.

**Comment by musikBear** [comment]
@Gabrielxd195 Im also a bit unsure what you mean, but if you ask for a way to create a Ghost-clipboard-image with a button in piano-roll, and a second piano-roll-button to set the clipboard-ghosts into a different piano-roll pattern, then its a good idea! 
Since this would take place inside a piano-roll-pattern, it could even be made so **only selected notes** would go into the Ghost-clipboard-image. If its done that way, it adds even more flexibility to the method -More LEGO-strategy :)

**Comment by Gabrielxd195** [comment]
> Are you asking for a way to show/hide the ghost notes in piano roll? If so that should be one toggleable button rather than two.

Not exactly, it doesn't have to do to show or hide. What I am referring to is a button that does something similar to '' refresh ''. when you enter and modify the pattern that you set as a ghost, you just press that button to `` update '' the image of the ghost notes, without leaving piano roll.

**Comment by Gabrielxd195** [comment]
> Ah wait, you mean you want to set the currently opened pattern as ghost notes, don't you? I still think that could be a single button, though.

That's what I mean, set it with the open pattern. but it is necessary to place a button for this to make it more visual.

**Comment by Gabrielxd195** [comment]
> @Gabrielxd195 Im also a bit unsure what you mean, but if you ask for a way to create a Ghost-clipboard-image with a button in piano-roll, and a second piano-roll-button to set the clipboard-ghosts into a different piano-roll pattern, then its a good idea!
> Since this would take place inside a piano-roll-pattern, it could even be made so **only selected notes** would go into the Ghost-clipboard-image. If its done that way, it adds even more flexibility to the method -More LEGO-strategy :)

That is not specifically my idea. wow !! but you just got an idea from another idea, this would be great! I recommend you make a thread with this idea to be taken into account.

**Comment by OutbackMan** [comment]
In my version of lmms 1.2.1, I don't have this 'set as ghost in piano-roll' option. Is it available only from compiling from source?

**Comment by Gabrielxd195** [comment]
> En mi versión de lmms 1.2.1, no tengo esta opción 'establecer como fantasma en piano-roll'. ¿Está disponible solo desde la compilación de la fuente?

No friend, unfortunately only the master version has this feature. and it doesn't have an installer binary.

**Comment by OutbackMan** [comment]
I compiled master and now I have ghost notes!

**Comment by allejok96** [comment]
>a way to show/hide the ghost notes in piano roll

> set the currently opened pattern as ghost notes

@Spekular, couldn't we have both? "Toggle ghost notes" and "Set as ghost notes". The later one could be restricted to selection...

Atm, opening up piano roll you see the Clear ghost note button, but no way to create them. Having to move out to song editor and right click a track is not the first thing you think about.
